torchvision/models/_api.py

```diff
@@ -75,7 +78,9 @@ def __getattr__(self, name):
 
 def get_weight(name: str) -> WeightsEnum:
     """
-    Gets the weight enum value by its full name. Example: "ResNet50_Weights.IMAGENET1K_V1"
+    Gets the weights enum value by its full name. Example: "ResNet50_Weights.IMAGENET1K_V1"
```

Comment:
Do we really want this and the rest to be Beta? I feel like we should reserve that for exceptional cases only. What is the plan to graduate these to stable?

Reply:
I would rather release this as beta and depending on community feedback graduate before release than releasing to stable and then having to communicate it as beta on the last moment.

Reply:
Do we have a way to remember to graduate this before the release?

I'm a bit concerned that we're going to start merging everything as Beta as soon as we're not 100% sure of it. Maybe it's a sign we need to think more about the PR?  Eventually, this might lead to the Beta status having no meaningful definition.

Reply:
IMO we should use beta for things where we're clearly not sure, or where we foresee potentially mixed community feedback. Is this the case here?

Also, we should consider that in the future, the guideline might be that Beta APIs  must come with a warning (which can potentially be disabled): https://github.com/pytorch/pytorch/issues/72317, https://github.com/pytorch/vision/pull/6173, https://github.com/pytorch/pytorch/compare/master...albanD:feature_warning_system?expand=1.
I feel like this would be overkill here?

Reply:
> IMO we should use beta for things where we're clearly not sure, or where we foresee potentially mixed community feedback. Is this the case here?

Yes that's totally the case. As you can see from #6330, our team had different opinions on the naming of the methods. We should seek the input of the community.

> I feel like this would be overkill here?

Yes I think adding warnings is too extreme and it's something that needs to be discussed separately.

Reply:
If we're only concerned about the naming, then I don't think breaking user code for changing an imperfect name to another one is something reasonable to do.

Reply:
Other than the naming, there are a few more API considerations that might arise such as the way the `module` parameter of `list_models()` works and Aligning the two registration APIs across Datasets and Models. More issues might pop up after we communicate the mechanism (I'm conscious that we all have blindspots). Overall I would much rather get input from the community on this than later introduce breaking changes in order to act upon their feedback. I think what we can do is try to push this mechanism via social and blogposts and collect feedback prior the release. If all is good, we should be able to promote to stable pretty fast aka before the next release.